# HG changeset patch
# User David Bateman
# Date 1202817645 -3600
#      Tue Feb 12 13:00:45 2008 +0100
# Node ID feb791af5d9ff4c1bcd6f0d4502849280d0ea862
# Parent  2d8315dcd8d2794db8f0df64c499b39357aaced8
Allow cast to treat the logical types

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,12 +1,13 @@
 2008-02-12  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m: Set pm3d implict to fix colorbars on
 	contour plots.
+	* miscellaneous/cast.m: Also treat the logical type.
 
  	* plot/__go_draw_axes__.m: Use the cdatamapping property to set
 	the cbrange.
 	* plot/__img__.m: Set cdatamapping to "direct".
 	* plot/imagesc.m: Set cdatamapping to "scaled".
 
 2008-02-08  Julien Pommier  <pommier@pianoteq.com>
 
diff --git a/scripts/miscellaneous/cast.m b/scripts/miscellaneous/cast.m
--- a/scripts/miscellaneous/cast.m
+++ b/scripts/miscellaneous/cast.m
@@ -25,17 +25,17 @@
 ## Author: jwe
 
 function retval = cast (val, typ)
 
   if (nargin == 2)
     if (ischar (typ))
       if (any (strcmp (typ, {"int8"; "uint8"; "int16"; "uint16";
 			     "int32"; "uint32"; "int64"; "uint64";
-			     "double"; "single"})))
+			     "double"; "single"; "logical"})))
 	retval = feval (typ, val);
       else
 	error ("cast: type name `%s' is not a built-in type", typ);
       endif
     else
       error ("cast: expecting type name as second argument");
     endif
   else
